### PR TITLE
Source map sources should to be relative to site.source

### LINKED ISSUE #115: sourcemap paths don't take --source directory into account
Running:
```
$ jekyll build -s <source> -d <dest>
```
yields a source map file with CWD-relative paths, rather than `<source>` relative paths as one would expect

### PR DESCRIPTION
**Pull request by ntkme**
`SassC` computes the "sources" path relative to `filename`, which is interpreted by `SassC` as relative to `Dir.pwd`. The problem is, when running `jekyll build -s site_source`, the `site_source` may not be equal to `Dir.pwd`, therefore the "sources" in the source map will end up with a relative path from `Dir.pwd` rather than `site_source`.

This PR fixes that by first compute the `filename`, `output_path`, and `source_map_file` in relative to `source_site` as relative to `Dir.pwd`. Because the `sourceMapUrl` in css and `file` in sourceMap are resolved against each other as relative path, the result is not different from before this patch. However, when generating `sources` paths in sourceMap, this will now be relative to `source_map_file`, which contains a relative path to `source_site` from `Dir.pwd`, therefore when relative paths of `sources` are resolved from `source_map_file`, the relative path to from `Dir.pwd` and `source_site` will cancels out  with each other, and finally become a relative path from `source_site`.

Closes #115.

### CODE REVIEW

**Comment by ntkme** [comment]
Although #115 not really affect functionality of source map when source is embedded, it has a security implication of revealing the system directory structure of side of `site.source`. Let's say we run this command:

``` sh
cd /tmp && jekyll serve -s /home/user/blog
```

The "sources" in source map will leak information about the directory structure in between `/tmp` and `site.source`, for example:

```
{
    "version": 3,
    "sources": ["../home/user/blog/_sass/test.scss", ...
```

This PR make sure that sources are always relative to `site.source`.

```
{
    "version": 3,
    "sources": ["_sass/test.scss", ...
```

File: lib/jekyll/converters/scss.rb
Comment (ashmaroli) [inline lib/jekyll/converters/scss.rb:291]: This needs to be memoized to avoid creating new `Pathname` instances on every call.
Additionally, instead of converting the pathname to a string and then subsequently use `File.join ...`, it'd be better to leave this as a pathname and then use `Pathname#join`:
```ruby
site_source_relative_from_pwd.join(sass_page.basename + '.css')
```

**Review by ashmaroli — CHANGES_REQUESTED** [review]
@ntkme Thank you for submitting this patch.
First of all, could you add a couple of tests for this?
Once there's test coverage, you may refactor confidently as suggested in my review.

File: lib/jekyll/converters/scss.rb
Comment (ntkme) [reply lib/jekyll/converters/scss.rb:291]: Agreed with memorizing it. However, `Pathname#join` would actually return a new instance of `Pathname`, that in the end I think won't be more efficient than `File.join`? And in the end the function is supposed to return a string anyways.

Comment (ashmaroli) [reply lib/jekyll/converters/scss.rb]: I can confirm your thoughts. I ran some benchmarks.
`File.join` was 13x faster than `Pathname#join`

File: lib/jekyll/converters/scss.rb
Comment (ashmaroli) [inline lib/jekyll/converters/scss.rb:291]: This line looks too long. Please break it up into multiple lines:
```ruby
      def site_source_relative_from_pwd
        @site_source_relative_from_pwd ||= \
          Pathname.new(site_source).relative_path_from(
            Pathname.new(Dir.pwd)
          ).to_s
      end
```

**Review by ashmaroli — CHANGES_REQUESTED** [review]
Thank you for memoizing the private method. Left a comment regarding line-length.

However, the test included is not adequate. It doesn't clarify the fact that `site.source` is not `Dir.pwd` (default). You'll have to either initialize a new `Jekyll::Site` instance with a *non-default "source"* or piggyback on existing tests that override default source.

**Comment by ntkme** [comment]
Modified the test to explicitly run `site.process` inside a `tmpdir` so that `site.source` is not equal to `Dir.pwd`.

**Review by ashmaroli — APPROVED** [review]

**Comment by ashmaroli** [comment]
@ntkme I have updated the test to ensure sourcemap contains what is needed and doesn't contain what isn't needed.
For future reference, don't bother rebasing after every change, we always squash commits on merge.

This will be merged after an approval from one more maintainer.
Thank you for the time spent on this.

**Review by DirtyF — APPROVED** [review]
LGTM

**Comment by ashmaroli** [comment]
Thank you @ntkme 
@jekyllbot: merge +fix

**Pull request merged**
